### PR TITLE
Undo and Redo Shortcuts!

### LINKED ISSUE #270: Ctrl+Z Ctrl+Y for Proof and Draw modes
Undo stuff, redo stuff

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by RyanR712** [comment]
Functionality is correct from my testing. Organization is probably not optimal. Particular attention would be appreciated for testing a bunch of different combinations of proof and draw tools. Marking ready for review

File: src/AEG-IO.ts [hunk 1]
```diff
@@ -45,21 +46,21 @@ interface atomObj {
  */
 interface proofNodeObj {
     tree: sheetObj;
-    appliedRule: string;
+    appliedRule: ProofModeMove;
```
Comment (James-Oswald) [inline src/AEG-IO.ts:49]: This is a massive API break, does loading files work, how doe the Enums get stored and loaded from json?

**Review by James-Oswald — COMMENTED** [review]
I have some large concerns regarding a potential API break that will prevent files from being saved and loaded.

File: src/AEG-IO.ts (hunk 1 above)
Comment (RyanR712) [reply src/AEG-IO.ts:48]: Discussed in the PMH server on 3/21

**Review by James-Oswald — APPROVED** [review]

**Pull request merged**